v2/internal/pkg/release/signature.go

```diff
@@ -65,125 +65,134 @@ func (o SignatureSchema) GenerateReleaseSignatures(ctx context.Context, images [
+
+				o.Log.Trace("field isEncrypted %v", md.IsEncrypted)
+				o.Log.Trace("field EencryptedToKeyIds %v", md.EncryptedToKeyIds)
+				o.Log.Trace("field IsSymmetricallyEncrypted %v", md.IsSymmetricallyEncrypted)
+				o.Log.Trace("field DecryptedWith %v", md.DecryptedWith)
+				o.Log.Trace("field IsSigned %v", md.IsSigned)
+				o.Log.Trace("field SignedByKeyId %v", md.SignedByKeyId)
+				o.Log.Trace("field SignedBy %v", md.SignedBy)
+				o.Log.Trace("field LiteralData %v", md.LiteralData)
+				o.Log.Trace("field SignatureError %v", md.SignatureError)
+				o.Log.Trace("field Signature %v", md.Signature)
+				// o.Log.Trace("field SignatureV3 %v", md.SignatureV3.IssuerKeyId)
+				// o.Log.Trace("field SignatureV3 %v", md.SignatureV3.CreationTime)
+
+				if md.Signature != nil {
+					if md.Signature.SigLifetimeSecs != nil {
+						expiry := md.Signature.CreationTime.Add(time.Duration(*md.Signature.SigLifetimeSecs) * time.Second)
+						if time.Now().After(expiry) {
+							o.Log.Debug("signature expired on %v ", expiry)
+						}
 					}
+				} else if md.SignatureV3 == nil {
+					return []v2alpha1.CopyImageSchema{}, fmt.Errorf("[GenerateReleaseSignatures] unexpected openpgp.MessageDetails: neither Signature nor SignatureV3 is set for %s image %s", digest, img.Source)
 				}
-			} else if md.SignatureV3 == nil {
-				return []v2alpha1.CopyImageSchema{}, fmt.Errorf("[GenerateReleaseSignatures] unexpected openpgp.MessageDetails: neither Signature nor SignatureV3 is set for %s image %s", digest, img.Source)
-			}
 
-			// write signature to cache
-			newImgSpec, err := image.ParseRef(img.Source)
-			if err != nil {
-				return []v2alpha1.CopyImageSchema{}, fmt.Errorf("[GenerateReleaseSignatures] could not parse identity docker reference image %v", err)
-			}
-			sigFilePath := fmt.Sprintf("%s%s/%s-sha256-%s", o.Opts.Global.WorkingDir, SignatureDir, newImgSpec.Tag, digest)
-			if _, err := os.Stat(sigFilePath); err != nil {
-				ferr := os.WriteFile(sigFilePath, data, 0644)
-				if ferr != nil {
-					return []v2alpha1.CopyImageSchema{}, fmt.Errorf("[GenerateReleaseSignatures] writing %v", ferr)
+				// write signature to cache
+				newImgSpec, err := image.ParseRef(img.Source)
+				if err != nil {
+					return []v2alpha1.CopyImageSchema{}, fmt.Errorf("[GenerateReleaseSignatures] could not parse identity docker reference image %w", err)
+				}
+				sigFilePath := fmt.Sprintf("%s%s/%s-sha256-%s", o.Opts.Global.WorkingDir, SignatureDir, newImgSpec.Tag, digest)
+				if _, err := os.Stat(sigFilePath); err != nil {
+					ferr := os.WriteFile(sigFilePath, data, 0600)
+					if ferr != nil {
+						return []v2alpha1.CopyImageSchema{}, fmt.Errorf("[GenerateReleaseSignatures] writing %w", ferr)
+					}
 				}
+				imgs = append(imgs, img)
+				data = nil
+			} else {
+				return []v2alpha1.CopyImageSchema{}, fmt.Errorf("[GenerateReleaseSignatures] no signature found for %s image %s", digest, img.Source)
```

Comment:
And since you're at it, you can also remove this whole `if-else` block by doing:
```suggestion
            if len(data) == 0 {
				return []v2alpha1.CopyImageSchema{}, fmt.Errorf("[GenerateReleaseSignatures] no signature found for %s image %s", digest, img.Source)
			}
```